src/main/java/com/redhat/cloud/policies/app/rest/PolicyCrudService.java

```diff
@@ -800,8 +816,16 @@ public Response getTriggerHistoryForPolicy(@PathParam("id") UUID policyId) {
      if (policy==null) {
        builder = Response.status(Response.Status.NOT_FOUND);
      } else {
+
        try {
-         String alerts = engine.findLastTriggered(policyId.toString(), false, user.getAccount());
+         Pager pager = PagingUtils.extractPager(uriInfo);
+
+         int limit = pager.getLimit();
+         limit = min(limit,200);
+         int pageNum = pager.getOffset() / limit;
+
+         String alerts = engine.findLastTriggered(policyId.toString(), false, pageNum, limit,
```

Comment:
I'll need a "paginated response", e.g. to know the total number of elements to decide if "next page" button is disabled or not.

Reply:
Yes. I am not sure yet how to get that from the engine